### ISSUE TITLE
request for pixelsnail pretrained .pt files

### ISSUE THREAD

**Issue by robot0321** [vvvm23/vqvae-2#7]
Hi @vvvm23, 
I saw your code today, it is novel and inspiring an I benefited a lot from it.
I'm sorry to bother you, but could you upload the .pt file you learned with pixel snail? 
Since I want to test the decoder alone, so I followed your very kind instructions.
but it takes a very long time because my GPU performance is not enough. :sweat_smile:

Thanks for your time!

**Comment by vvvm23** [comment]
Hi, I also didn't have the resources to train the PixelSnail component at the time, so no file exists.

I am actively working on a refactor of the repository with weights, so should have weights available after that (maybe not PixelSnail, but something that does the same thing)

**Comment by shibbit** [comment]
You don't _necessarily_ need the pixelCNN pretrained to test the decoder, however you will need it to produce the best results(otherwise they will be blurry), because in my opinion which might not be right, the main target to insert pixelCNN is to add more variety to the generated results. The biggest difference between VQ-VAE and vanilla VAE is that the pixelCNN used here is designed to fit a _prior_  distribution of the codebook which is low-dimention representation of the input, whereas the vanilla VAE simply presume the prior is Gaussian, both these priors are used to control the generated results.  However, training the autoregressive model like pixelCNN takes enormous  computation cost, thus the VQ-VAE combine with codebook and pixelCNN together to make this whole thing feasible, and did improve the image quality a lot. 
So, to summarize, if your task is GENERATION which takes variety and image quality as the targets, then you will need the pretrained pixelCNN, but if your task is something like RECONSTRUCTION or DENOISING, etc. then you can just directly inference but you have to designate the inputs.
